src/nix/network-proxy.hh

```diff
@@ -0,0 +1,27 @@
+#pragma once
+///@file
+
+#include "types.hh"
+
+namespace nix {
+
+/**
+ * Environment variables relating to network proxying. These are used by
+ * a few misc commands.
+ */
+constexpr auto networkProxyVariables = {
+    // FIXME until https://github.com/llvm/llvm-project/issues/61560 is
+    // implemented
+    // clang-format off
+    "http_proxy",
+    "https_proxy",
+    "ftp_proxy",
+    "all_proxy",
+    "HTTP_PROXY",
```

Comment:
https://curl.se/docs/manpage.html weird it says it is not a thing, but we can keep for back compat. Just add a comment maybe.